pkg/download/download.go

```diff
@@ -18,17 +18,37 @@ import (
 
 // Decompressed downloads and decompresses a file from a URL to a local path.
 //
-// It supports gz, bz2 and gz compressed files.
+// It supports gz, bz2 and xz compressed files.
 func Decompressed(client *http.Client, u url.URL, path string) error {
-	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+
+	// atomically write to file
+	dir, file := filepath.Split(path)
+	if dir == "" {
+		// If the file is in the current working directory, then dir will be "".
+		// However, this means that ioutil.TempFile will use the default directory
+		// for temporary files, which is wrong.
+		dir = "."
+	}
+
+	// ensure dir exists
+	if err := os.MkdirAll(dir, 0o755); err != nil {
 		return err
 	}
-	tmpFile, err := ioutil.TempFile("", fmt.Sprintf("%s*", filepath.Base(path)))
+
+	tmpFile, err := ioutil.TempFile(dir, file)
 	if err != nil {
-		return err
+		return fmt.Errorf("create temporary file: %w", err)
 	}
 	defer tmpFile.Close()
 
+	// Clean up tmp file if not moved
+	moved := false
+	defer func() {
+		if !moved {
+			os.Remove(tmpFile.Name())
```

Comment:
Nit/Question:
```go
tmpFile.Close() // should this be called first?
os.Remove(tmpFile.Name())
```

Reply:
afaik, removing a file is filesystem level operation. A file can be can be opened multiple times by different/same processes, and can be removed without closing the file. In this case, I don't think the order of remove and close matters.

Reply:
Wrote a small snippet of code in https://go.dev/play/p/Iy5L8VILwvK to test that removing a file before closing also doesn't result in any errors.